### PR TITLE
fixed SyntaxError with trailing zeros

### PR DESCRIPTION
**Pull request by haansn08**

### CODE REVIEW

File: test/ast.spec.js [hunk 1]
```diff
@@ -466,6 +466,11 @@ describe('AST', () => {
             expect(getParsedSql(sql)).to.equal('SELECT NULL');
         });
 
+        it('should support trailing zeros', function() {
```
Comment (mbaumgartl) [inline test/ast.spec.js]: Can you replace the function declaration by an arrow function?

File: test/ast.spec.js [hunk 2]
```diff
@@ -466,6 +466,11 @@ describe('AST', () => {
...
+            expect(getParsedSql('SELECT 042')).oneOf(['SELECT 42', 'SELECT 042']);
```
Comment (mbaumgartl) [inline test/ast.spec.js]: I'm curious: what is the use case for this?

Why do you test the result w/ and w/o leading zeros? Shouldn't it be always the same result?

File: test/ast.spec.js (hunk 1 above)
Comment (haansn08) [reply test/ast.spec.js]: Done. However, [passing arrow functions to Mocha is discouraged](https://mochajs.org/#arrow-functions).

File: test/ast.spec.js (hunk 2 above)
Comment (haansn08) [reply test/ast.spec.js]: I think the parser should be allowed to either drop the leading zeros or leave them as they are. Form my understanding both are correct behavior.

Comment (mbaumgartl) [reply test/ast.spec.js]: The parser must produce the same result for the same input. In this case it should remove the leading zero. Please replace `oneOf` with `equal` and test for number w/o leading zeros.

Comment (haansn08) [reply test/ast.spec.js]: Done.

**Pull request merged**